src/providers/filters/deep.ts

```diff
@@ -9,7 +9,7 @@ export default class DeepFilter {
 		const maxPatternDepth = this._getMaxPatternDepth(positive);
 		const negativeRe = this._getNegativePatternsRe(negative);
 
-		return (entry) => this._filter(basePath, entry, negativeRe, maxPatternDepth);
+		return (entry) => this._filter(basePath, entry as Entry, negativeRe, maxPatternDepth);
```

Comment:
JFYI: When we use the type assertion mechanism we introduce unsafe code. For example, then inside this function we can refer to the `base` property, which may not be accessible (`undefined`).

Reply:
Yes I know, but if not so I have to change `fsWalk.Entry` or `DeepFilterFunction`.